mesonbuild/build.py

```diff
@@ -669,10 +669,25 @@ def get_all_link_deps(self):
         return self.get_transitive_link_deps()
 
     def get_transitive_link_deps(self):
```

Comment:
A simpler way to cache this would also be to add `@functools.lru_cache()` as a decorator here!

Reply:
Thats a neat helper, thx! :)